### PR TITLE
Add integration test and documentation for Stackdriver Logging

### PR DESCRIPTION
**Pull request by cassand**

### CODE REVIEW

File: tomcat/src/main/tomcat-base/conf/server.xml [hunk 1]
```diff
@@ -57,7 +57,11 @@
 
         &gcp-configuration;
 
-        &distributed-session-configuration;
+        <Context path="/">
+
+            &distributed-session-configuration;
```
Comment (meltsufin) [inline tomcat/src/main/tomcat-base/conf/server.xml]: This change doesn't seem relevant to the PR

File: README.md [hunk 2]
```diff
@@ -49,6 +49,19 @@ COPY your-application.war ROOT.war
 The Tomcat instance can be configured through the environment variable `TOMCAT_PROPERTIES` which is
 a comma-separated list of `name=value` pairs appended to `catalina.properties`.
 
+### Stackdriver Logging
+When the Tomcat runtime is running on Google App Engine, all output to stdout/stderr is captured by Stackdriver
```
Comment (meltsufin) [inline README.md]: "Google App Engine" -> "Google App Engine flexible environment"
"captured" -> "forwarded to Stackdriver Logging and available in the Cloud Console Log Viewer"

File: README.md [hunk 3]
```diff
@@ -49,6 +49,19 @@ COPY your-application.war ROOT.war
 The Tomcat instance can be configured through the environment variable `TOMCAT_PROPERTIES` which is
 a comma-separated list of `name=value` pairs appended to `catalina.properties`.
 
+### Stackdriver Logging
+When the Tomcat runtime is running on Google App Engine, all output to stdout/stderr is captured by Stackdriver 
+and is available via the Stackdriver logging console.
+ 
+However more detailed and integrated logs are available if the Stackdriver logging mechanism is used directly.
```
Comment (meltsufin) [inline README.md]: This is too vague to decide whether to use it. See https://github.com/GoogleCloudPlatform/jetty-runtime#enhanced-stackdriver-logging-beta for a better description.

File: README.md [hunk 4]
```diff
@@ -49,6 +49,19 @@ COPY your-application.war ROOT.war
...
+
+To take advantage of this integration, add the [Google Cloud Java Client for Logging](https://github.com/GoogleCloudPlatform/google-cloud-java/tree/master/google-cloud-logging) 
+to your dependencies and provide a `logging.properties` file as part of the resources of the application with the following content:
```
Comment (meltsufin) [inline README.md]: Be more specific about where to put this file and that it's a Java Util Logging configuration file.

File: README.md [hunk 5]
```diff
@@ -49,6 +49,19 @@ COPY your-application.war ROOT.war
...
+ 
+```properties
+handlers=com.google.cloud.logging.LoggingHandler
```
Comment (meltsufin) [inline README.md]: Give a full example like in the Jetty docs.

File: tests/test-war-integration/pom.xml [hunk 6]
```diff
@@ -45,8 +45,51 @@
...
+    <dependency>
+      <groupId>com.google.cloud</groupId>
+      <artifactId>google-cloud-logging</artifactId>
+      <version>1.2.1</version>
+    </dependency>
   </dependencies>
 
+  <profiles>
+    <profile>
+      <id>local-docker-build</id>
```
Comment (meltsufin) [inline tests/test-war-integration/pom.xml:57]: Where is this profile used? Please put a comment explaining what it's for.

File: tests/test-war-integration/src/main/resources/logging.properties [hunk 7]
```diff
@@ -0,0 +1,4 @@
+com.google.cloud.runtimes.level = FINE
+com.google.cloud.logging.LoggingHandler.level = FINE
+com.google.cloud.logging.LoggingHandler.log = appengine.googleapis.com%2Fstdout
+handlers=com.google.cloud.logging.LoggingHandler,java.util.logging.ConsoleHandler
```
Comment (meltsufin) [inline tests/test-war-integration/src/main/resources/logging.properties]: Why both handlers?

**Review by meltsufin — CHANGES_REQUESTED** [review]

File: tomcat/src/main/tomcat-base/conf/server.xml [hunk 8]
```diff
@@ -57,11 +57,7 @@
 
         &gcp-configuration;
 
-        <Context path="/">
-
-            &distributed-session-configuration;
-
-        </Context>
+        &distributed-session-configuration;
```
Comment (meltsufin) [inline tomcat/src/main/tomcat-base/conf/server.xml]: This file shouldn't be modified at all, right?

File: README.md [hunk 9]
```diff
@@ -49,6 +49,26 @@ COPY your-application.war ROOT.war
 The Tomcat instance can be configured through the environment variable `TOMCAT_PROPERTIES` which is
 a comma-separated list of `name=value` pairs appended to `catalina.properties`.
 
+### Stackdriver Logging
```
Comment (meltsufin) [inline README.md]: Can you move it under "Optional Modules" and rename "Optional Modules" to "Optional Features"?

File: tomcat/src/main/tomcat-base/conf/server.xml (hunk 8 above)
Comment (cassand) [reply tomcat/src/main/tomcat-base/conf/server.xml]: It isn't modified, this is a revert to the original file (e.g the change is visible for the commit but not in the PR).

File: tests/test-war-integration/pom.xml (hunk 6 above)
Comment (cassand) [reply tests/test-war-integration/pom.xml:57]: The profile is activated by default, it is disabled during the `cloudbuild` step (to avoid creating a Docker image locally). This profile is already present in the other test applications.

**Review by meltsufin — APPROVED** [review]
